src/finch/tensor.py

```diff
@@ -16,17 +16,53 @@ def __str__(self):
 
 
 class Tensor(_Display):
-    def __init__(self, lvl=None, arr=None, jl_data=None):
+    """
+    A wrapper class for Finch.Tensor and Finch.SwizzleArray
+    """
+    row_major = "C"
+    column_major = "F"
+
+    @classmethod
+    def preprocess_order(
+        cls, order: Union[str, tuple[int, ...], None], ndim: int
+    ) -> tuple[int, ...]:
+        if order == cls.column_major:
+            permutation = tuple(range(1, ndim + 1))
+        elif order == cls.row_major or order is None:
+            permutation = tuple(range(1, ndim + 1)[::-1])
+        elif isinstance(order, tuple):
+            if min(order) == 0:
+                order = tuple(i + 1 for i in order)
+            if (
+                len(order) == ndim and
+                all([i in order for i in range(1, ndim + 1)])
+            ):
+                permutation = order
+            else:
+                raise ValueError(f"Custom order is not a permutation: {order}.")
+        else:
+            raise ValueError(f"order must be 'C', 'F' or a tuple, but is: {type(order)}.")
+
+        return permutation
+
+
+    def __init__(self, lvl=None, arr=None, jl_data=None, order=None):
         if arr is not None and not isinstance(arr, np.ndarray):
-            raise ValueError("For now only numpy input allowed")
+            raise ValueError("For now only numpy input allowed.")
 
+        # constructor for levels description and NumPy array input
         if lvl is not None and arr is not None and jl_data is None:
-            self._obj = jl.Tensor(lvl._obj, np.array(arr, order="F"))
-        elif jl_data is not None:
+            order = self.preprocess_order(order, arr.ndim)
+            inv_order = tuple(i - 1 for i in jl.invperm(order))
+            self._obj = jl.swizzle(jl.Tensor(lvl._obj, arr.transpose(inv_order)), *order)
+        # constructor for a raw julia object
+        elif jl_data is not None and lvl is None and arr is None:
+            if order is not None:
+                raise ValueError("When passing a julia object order can't be provided.")
```

Comment:
maybe also check that the object is a swizzle of tensor?

Reply:
Done! I added another check.